client/config/toml.go

```diff
@@ -38,13 +52,66 @@ func writeConfigToFile(configFilePath string, config *ClientConfig) error {
 		return err
 	}
 
+	// Check if config file exists
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		fmt.Println("creating new node configuration in", filepath.Dir(filepath.Dir(configFilePath)))
+	}
 	if err := configTemplate.Execute(&buffer, config); err != nil {
 		return err
 	}
 
 	return os.WriteFile(configFilePath, buffer.Bytes(), 0o600)
 }
 
+// HomeDirConfig is a struct that contains the home directory and defines the mapping
+// for the TOML writer.
+type HomeDirConfig struct {
+	HomeDir string `mapstructure:"home-dir" json:"home-dir"`
+}
+
+// WriteHomeDirToFile parses homeDirTemplate, renders the template with the given new home directory
+// and writes it to the given file.
+func WriteHomeDirToFile(filePath, homeDir string) error {
+	var buffer bytes.Buffer
+
+	tmpl := template.New("homeDirFileTemplate")
+	homeTemplate, err := tmpl.Parse(defaultHomeTemplate)
+	if err != nil {
+		return err
+	}
+
+	homeDirConfig := HomeDirConfig{HomeDir: homeDir}
+	if err := homeTemplate.Execute(&buffer, homeDirConfig); err != nil {
+		return err
+	}
+	return os.WriteFile(filePath, buffer.Bytes(), 0o600)
+}
+
+// ReadHomeDir tries to return the currently stored home directory from the
+// given config directory.
+func ReadHomeDir(configPath string, v *viper.Viper) (string, error) {
+	homeDirConfig := HomeDirConfig{}
+	v.AddConfigPath(configPath)
+	v.SetConfigName("home")
+	v.SetConfigType("toml")
+
+	homeFilePath := filepath.Join(configPath, "home.toml")
+	if _, err := os.Stat(homeFilePath); os.IsNotExist(err) {
+		folder := filepath.Dir(configPath)
+		if err := WriteHomeDirToFile(homeFilePath, folder); err != nil {
+			return "", err
+		}
+	}
```

Comment:
Ditto about handling other non-nil errors here. I'd highly advise against force creating a home file path in a function whose name is ReadHomeDir and tries to get the currently stored home directory. At least please rename it to ReadOrMakeHomeDir and add a comment that it'll force create the directory if it doesn't exist.